storage.go

```diff
@@ -85,6 +85,10 @@ func (svc *StorageServiceOp) ListVolumes(ctx context.Context, params *ListVolume
 	if params != nil {
 		if params.Region != "" && params.Name != "" {
 			path = fmt.Sprintf("%s?name=%s&region=%s", path, params.Name, params.Region)
+		} else if params.Region != "" {
+			path = fmt.Sprintf("%s?region=%s", path, params.Region)
+		} else if params.Name != "" {
+			path = fmt.Sprintf("%s?name=%s", path, params.Name)
```

Comment:
This block _could_ be simplified with something like
```

path += "?"

if params.Region != "" {
	path = fmt.Sprintf("region=%s", path, params.Region)
} 

if params.Name != "" {
	path = fmt.Sprintf("name=%s", path, params.Name)
}
```

Reply:
What if region and name is not set though?

Reply:
Then there won't be a query string, but `example.com/quux?` is a valid URL....

Reply:
truee

Reply:
Ah, but you'd need to prefix each `%s` with `&` if you make this change...